pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -617,7 +627,9 @@ func (s *Controller) Cluster() cluster.ID {
 }
 
 // AppendServiceHandler adds service resource event handler. Service Entries does not use these handlers.
-func (s *Controller) AppendServiceHandler(_ model.ServiceHandler) {}
+func (s *Controller) AppendServiceHandler(h model.ServiceHandler) {
```

Comment:
https://github.com/istio/istio/blob/c8fdbc7b8a9275e02b9c7d6e5f888ed41b905a5d/pilot/pkg/model/controller.go#L67

We have a util class to dedupe most of the handler stuff.

Reply:
this was necessary so the service entry controller could fire events on new auto vip allocation